rawmemchar is not available on musl, guard it with defined(__GLIBC__)

--- a/memcheck/tests/str_tester.c	2021-10-10 14:17:32.000000000 +0000
+++ b/memcheck/tests/str_tester.c	2021-11-16 19:37:26.950006442 +0000
@@ -504,7 +504,7 @@
 #endif
 
 // DDD: better done by testing for the function.
-#if !defined(__APPLE__) && !defined(__sun) && !defined(__FreeBSD__)
+#if defined(__GLIBC__) && !defined(__APPLE__) && !defined(__sun) && !defined(__FreeBSD__)
 static void
 test_rawmemchr (void)
 {
@@ -1451,7 +1451,7 @@
   test_strchrnul ();
 # endif
 
-# if !defined(__APPLE__) && !defined(__sun) && !defined(__FreeBSD__)
+# if defined(__GLIBC__) && !defined(__APPLE__) && !defined(__sun) && !defined(__FreeBSD__)
   /* rawmemchr.  */
   test_rawmemchr ();
 # endif
